crates/sdk/src/client.rs

```diff
@@ -0,0 +1,126 @@
+use crate::{
+    local::{LocalProver, LocalProverBuilder},
+    network::{NetworkProver, NetworkProverBuilder},
+    network_v2::DEFAULT_PROVER_NETWORK_RPC,
+    opts::ProofOpts,
+    proof::SP1ProofWithPublicValues,
+    prover::Prover,
+};
+use anyhow::Result;
+use sp1_core_executor::{ExecutionError, ExecutionReport};
+use sp1_core_machine::io::SP1Stdin;
+use sp1_primitives::io::SP1PublicValues;
+use sp1_prover::{SP1ProvingKey, SP1VerifyingKey};
+use std::env;
+
+pub struct None;
+
+pub struct ProverClient {
+    inner: Box<dyn Prover>,
+}
+
+pub struct ProverClientBuilder<T> {
+    inner_builder: T,
+}
+
+impl ProverClient {
+    pub fn builder() -> ProverClientBuilder<None> {
+        ProverClientBuilder { inner_builder: None }
+    }
+
+    #[deprecated(note = "Use ProverClient::builder() instead")]
+    pub fn new() -> Self {
+        Self::create_from_env()
+    }
+
+    fn create_from_env() -> Self {
```

Comment:
So I think that the DynBuilder should also expose default_{cycle_limt, timeout}?

I think that we should remove this method "create_from_env" and make the method like "build_with_env" that than overrides stuff at the end

Reply:
do you mean as methods? Currently I have:
```
/// The default timeout seconds for a proof request to be generated (4 hours).
pub const DEFAULT_TIMEOUT: u64 = 14400;

/// The default cycle limit for a proof request.
pub const DEFAULT_CYCLE_LIMIT: u64 = 100_000_000;
```

Reply:
Yaa I think since these are common defaults we can optionally let the user set them at builder level wdyt? Otherwise default to these